src/common/util/firebase.ts

```diff
@@ -126,9 +126,9 @@ export function saveGame(game: w.SavedGame): firebase.database.ThenableReference
 
 // Cards and text
 
-export function listenToRecentCards(callback: (data: any) => any): void {
+export function listenToRecentCards(callback: (data: any) => any, uid?: string): void {
```

Comment:
Hmm, actually I think the `uid` parameter should come first. It's a little gross to have the callback not be the last argument, you know?

Reply:
You can make `uid` be of type `string | null` and just check against `=== null`. (Just make sure to change all the invocations of `listenToRecentCards()`.)